src/files.ts

```diff
@@ -4,14 +4,24 @@
  */
 
 import { loadState } from '@nextcloud/initial-state'
-import { registerFileAction } from '@nextcloud/files'
+import { registerFileAction, getFileActions } from '@nextcloud/files'
 
 import { setupWebDavDecryptionProxy } from './services/webDavProxy.ts'
 import downloadUnencryptedAction from './services/downloadUnencryptedAction.ts'
+import logger from './services/logger.ts'
 
 const userConfig = loadState('end_to_end_encryption', 'userConfig', { e2eeInBrowserEnabled: false })
 
 if (userConfig.e2eeInBrowserEnabled) {
 	setupWebDavDecryptionProxy()
 	registerFileAction(downloadUnencryptedAction)
+	disableFileAction('download')
+	disableFileAction('move-copy')
 }
+
+function disableFileAction(actionId: string) {
+	logger.debug('Disabling file action', { actionId })
+	const actions = getFileActions()
+	const action = actions.find(action => action.id === actionId) as any
+	action._action.enabled = () => false
```

Comment:
We need to check whether the file is encrypted...

Reply:
Files do not have the `is-encrypted` info. I might need to share the state with the proxy to know whether the parent folder is encrypted or not.

Reply:
- [ ] Also need to disable the download unencrypted aciton for non encrypted files